src/goto-programs/restrict_function_pointers.cpp

```diff
@@ -193,6 +198,51 @@ void handle_call(
                        address_of_exprt{function_pointer_target_symbol_expr}}));
   }
 }
+
+void get_by_name_restriction(
+  const goto_functiont &goto_function,
+  const function_pointer_restrictionst::restrictionst &by_name_restrictions,
+  function_pointer_restrictionst::restrictionst &restrictions,
+  const goto_programt::const_targett &location)
+{
+  PRECONDITION(location->is_function_call());
+
+  const exprt &function = location->get_function_call().function();
+
+  if(!can_cast_expr<dereference_exprt>(function))
+    return;
+
+  auto const &function_pointer_call_site = to_symbol_expr(
+    to_dereference_expr(function)
+      .pointer());
```

Comment:
:question: Is it guaranteed that the `pointer()` result will be a symbol expr? I think it either needs a PRECONDITION or an early return as this isn't obvious to me

Reply:
Yes, this is guaranteed due to function pointer labelling having been run before. I think the precondition is redundant here as `to_symbol_expr()` (and all other such conversion functions) already include a precondition as their first statement.

Reply:
Fair - however, if I'm debugging this and I get an error saying isn't a symbol I don't have any clue why you'd expect this, whereas if you provide a specific one, you can add a comment saying must be symbol because pointer_labelling has been run (and therefore, hitting this suggests someone has forgot to run pointer labelling).